pandora-common/src/event.ts

```diff
@@ -6,6 +6,8 @@ export interface ITypedEventEmitter<T extends TypedEvent> {
 	getSubscriber(key: keyof T): (onStoreChange: () => void) => () => void;
 }
 
+export type TypedEventEmitterEvents<T> = T extends ITypedEventEmitter<infer E> ? E : never;
```

Comment:
Nit: Shouldn't this check that the `T` does extend `ITypedEventEmitter<any>` to catch errors early?

Reply:
I tried that but it did not work so that is why I did not add an extends
```
pandora-server-directory build$ pnpm run --silent clean && tsc -p ./tsconfig.json
│ src/networking/connection_client.ts(85,62): error TS2344: Type 'AccountToken' does not satisfy the constraint 'ITypedEventEmitter<any>'.
│   Types of property 'on' are incompatible.
│     Type '<K extends "extended" | "tokenDestroyed">(event: K, listener: (value: { tokenDestroyed: AccountToken; extended: AccountToken; }[K]) => void) => () => void' is not assignable to type '<K extends string | number | symbol>(s: K, listener: (v: any)…
│       Types of parameters 'event' and 's' are incompatible.
│         Type 'K' is not assignable to type '"extended" | "tokenDestroyed"'.
│           Type 'string | number | symbol' is not assignable to type '"extended" | "tokenDestroyed"'.
│             Type 'string' is not assignable to type '"extended" | "tokenDestroyed"'.
```